Client/mods/deathmatch/logic/rpc/CVehicleRPCs.cpp

```diff
@@ -653,3 +654,22 @@ void CVehicleRPCs::SetVehiclePlateText(CClientEntity* pSourceEntity, NetBitStrea
         }
     }
 }
+
+
+void CVehicleRPCs::SetVehicleNitroActivated(CClientEntity* pSourceEntity, NetBitStreamInterface& bitStream)
+{
+    bool state = bitStream.ReadBit();
+
+    CClientVehicle* vehicle = m_pVehicleManager->Get(pSourceEntity->GetID());
+    if (!vehicle)
+        return;          
+
+    if (!vehicle->IsNitroInstalled())
+        return;            
+
```

Comment:
It seems that there has to be IsVehicleNitroActivated check as well.

Reply:
I add this check rely on the client code method below it
```c++
 if ((vehicle->GetNitroLevel() < 0) == state)
     return;
``` 

```c++
bool CStaticFunctionDefinitions::IsVehicleNitroActivated(CClientVehicle& Vehicle, bool& bActivated)
{
    if (Vehicle.IsNitroInstalled())
    {
        // If nitro level < 0, nitro is activated. (until nitro level reaches -1, at that point it will become 0 and increase instead of decrease)
        bActivated = (Vehicle.GetNitroLevel() < 0);
        return true;
    }

    return false;
}

```

Reply:
By the way
```
bool bAlreadyActivated;
IsVehicleNitroActivated(pVehicle, bAlreadyActivated);
```
has undefined behaviour. Maybe someone desires to fix it.